### PR TITLE
[FIX] website: ensure websites have snippet filters

### PR DESCRIPTION
**Pull request by rdeodoo**
Since the feature was implemented with the default snippets having a website_id
set to the default website, it means that every other websites won't have
snippet filters.

This commit bootstrap filters for every new created website.

Note that in master, multi-website and generic filters are handled with
550d2935bd.

opw-2381423

### CODE REVIEW

**Comment by rdeodoo** [comment]
@JKE-be Can you have a first look to see if that's the way you want it to go?

The backport commit should be enough to fix when the user do a -u (its backport + set website_id to False)

For when no -u, the other commit should handle cases of creating a new website, removing the default website.

But one issue remain: User do not -u, delete the default website (filters are lost), then create a new website (no filters). 
Solution : Since no -u, we can't play with `ondelete` or `required` so maybe we prevent delete of default website while the user hasn't done -u, or we hardcode in python the creation of the default filters instead of copy, again only if no -u

**Comment by rdeodoo** [comment]
@fw-bot up to 14.0

File: addons/website_sale/static/src/snippets/s_dynamic_snippet_products/options.js [hunk 1]
```diff
@@ -20,16 +21,20 @@ const dynamicSnippetProductsOptions = s_dynamic_snippet_carousel_options.extend(
      */
     onBuilt: function () {
         this._super.apply(this, arguments);
+        let domain = wUtils.websiteDomain(this);
+        domain.push(['action_server_id.model_id.model', '=', 'product.product'])
```
Comment (JKE-be) [inline addons/website_sale/static/src/snippets/s_dynamic_snippet_products/options.js:25]: or filter_id.model :/

Comment (rdeodoo) [reply addons/website_sale/static/src/snippets/s_dynamic_snippet_products/options.js]: Indeed :+1:

**Comment by rdeodoo** [comment]
Should be ok now.
Tested every dynamic snippet with new/deleted website.

The only issue remaining is that the **demo** data `website2` has no website_sale filter. Indeed the website bootstraping code during -i is ignoring the website_sale module code.
But that can be ignored IMHO.

**Comment by rdeodoo** [comment]
As a reminder since those messages are folded, the forward port is disabled, so this is just for 14.0.
The fix typo commit will be cherry picked in master

**Comment by rdeodoo** [comment]
@JKE-be Fixed the right error for non admin user

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by JKE-be** [comment]
@robodoo rebase-ff

**Pull request closed without merging**